src/imgui/date.hpp

```diff
@@ -99,50 +99,35 @@ struct Month
   }
   [[nodiscard]] constexpr Year toYears() const
   {
-    return Year{ value / 12 };
+    return Year{ durationValue / 12 };
   }
   constexpr bool isOverAYear() const
   {
-    return value > 12;
+    return durationValue > 12;
   }
-  int value{};
+  int durationValue{};
 };
 
 template<typename T>
-struct isDurationType : std::false_type {};
-
-template<typename T>
-constexpr bool isDuration = isDurationType<T>::value;
-
-template<>
-struct isDurationType<Day> : std::true_type {};
-
-template<>
-struct isDurationType<Week> : std::true_type {};
-
-template<>
-struct isDurationType<Month> : std::true_type {};
-
-template<>
-struct isDurationType<Year> : std::true_type {};
-
-template<typename T>
-concept Duration = isDuration<T>;
+concept Duration = requires(T c)
+{
+  c.durationValue;
+};
```

Comment:
This change I don't like, because one day there may be another struct somewhere with a field called durationValue, and then this could cause a Weird compilation error. If you want to simplify this concept, we could make a base class for all of them and make the concept be "is derived from this class".

Reply:
I agree yes. I changed the field name because of that but indeed is not a strong guarantee. maybe going the inheritance  way is the way. I guess after that this is ready to review

Reply:
We're still missing some things that I think would be nice:
operator-(Date, Date) -> Day
separating duration from date value
month.toString()
putting everything in namespaces

Reply:
cool